### PR TITLE
fix: persist Toelichting when finishing a Taak

### PR DESCRIPTION
**Pull request by felixcicatt**
When clicking "Opslaan & afronden" for a Taak, the Toelichting is now correctly persisted.

Solves PZ-1570

### CODE REVIEW

File: src/main/app/src/app/taken/taak-view/taak-view.component.ts [hunk 1]
```diff
@@ -341,15 +341,6 @@ export class TaakViewComponent
     }
   }
 
-  partialEditTaak(value: string, field: string): void {
```
Comment (felixcicatt) [inline src/main/app/src/app/taken/taak-view/taak-view.component.ts:344]: This function was never called, so I removed it. This in turn was the only code resulting in a call to takenService.update, so it allowed for a method on te back end to be made private. See other comment.

File: src/main/java/net/atos/zac/app/taken/TakenRESTService.java [hunk 2]
```diff
@@ -260,18 +262,12 @@ public RESTTaak toekennenAanIngelogdeMedewerker(final RESTTaakToekennenGegevens
         return restTaakConverter.convert(task);
     }
 
-    @PATCH
-    public RESTTaak updateTaak(final RESTTaak restTaak) {
+    private Task updateTaak(RESTTaak restTaak) {
```
Comment (felixcicatt) [inline src/main/java/net/atos/zac/app/taken/TakenRESTService.java:265]: This endpoint was never called from the front end, so I made it a private method. I felt asserting policies and sending out events should be the responsibility of endpoints, so removed that code from this method.

File: src/main/java/net/atos/zac/app/taken/TakenRESTService.java [hunk 3]
```diff
@@ -165,11 +165,13 @@ public RESTTaak readTaak(@PathParam("taskId") final String taskId) {
...
     @Path("taakdata")
     public RESTTaak updateTaakdata(final RESTTaak restTaak) {
-        final Task task = takenService.readOpenTask(restTaak.id);
+        Task task = takenService.readOpenTask(restTaak.id);
         assertPolicy(getTaakStatus(task) != AFGEROND && policyService.readTaakRechten(task).wijzigen());
         taakVariabelenService.setTaakdata(task, restTaak.taakdata);
         taakVariabelenService.setTaakinformatie(task, restTaak.taakinformatie);
-        updateTaak(restTaak);
+        task = updateTaak(restTaak);
+        eventingService.send(TAAK.updated(task));
```
Comment (felixcicatt) [inline src/main/java/net/atos/zac/app/taken/TakenRESTService.java:173]: updateTaak is now a private method that only persists data, so the responsibility to send these two events now lies with this endpoint. See comment below.

Comment (bas-info-nl) [reply src/main/java/net/atos/zac/app/taken/TakenRESTService.java:173]: Method `updateTaak` is reloading the `Task` from the `takenService`, and doesn't do much with it. I would incorporate the couple of lines of code here instead.

**Review by bas-info-nl — APPROVED** [review]
Please check the comments, and fix. Looks good otherwise 😄

File: src/main/java/net/atos/zac/app/taken/TakenRESTService.java (hunk 2 above)
Comment (felixcicatt) [reply src/main/java/net/atos/zac/app/taken/TakenRESTService.java:265]: It was being used by updateTaakdata, which is why I made it private in stead of removing it. But the suggestion you make in your other comment might make it possible to remove it after all.

File: src/main/java/net/atos/zac/app/taken/TakenRESTService.java (hunk 3 above)
Comment (felixcicatt) [reply src/main/java/net/atos/zac/app/taken/TakenRESTService.java:173]: That's fair. It results in a very slight duplication of code between updateTaakdata and completeTaak but I think that's not an issue. Will do, thanks!

Comment (felixcicatt) [reply src/main/java/net/atos/zac/app/taken/TakenRESTService.java:175]: Done :)

**Pull request merged**